pkg/view/component/single_button.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright (c) 2021 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package component
+
+import (
+	"github.com/pkg/errors"
+	"github.com/vmware-tanzu/octant/internal/util/json"
+	"github.com/vmware-tanzu/octant/pkg/action"
+)
+
+// SingleButton is a component for a SingleButton
+// +octant:component
+type SingleButton struct {
```

Comment:
Just `Button`

Reply:
It collides with the definition on https://github.com/vmware-tanzu/octant/blob/master/pkg/view/component/button.go#L70
So, to avoid that, I'm gonna rename in that file the struct to ButtonElement to be able to rename this one to just `Button`

Reply:
So, I checking this change and I think it can be a little aggressive because in line https://github.com/vmware-tanzu/octant/blob/master/pkg/view/component/button.go#L94 we define the array of `Buttons`, if I change the structure to be `ButtonElement` the plugins are creating buttons groups like this are gonna break: 

```
					ButtonGroup: &ButtonGroup{
						Base: Base{},
						Config: ButtonGroupConfig{
							Buttons: []Button{{
								Name: "test",
							}},
						},
					},
```

Reply:
That would still be `[]Button` .. the `Button` struct itself, now needs to inheirt from base and you'll want a `ButtonConfig` type to represent the button options.

Reply:
Similar to this, https://gist.github.com/wwitzel3/82fb4d1e0b6dad1fdb9bb204ff03214c

Reply:
If we need to break existing APIs to ensure we have the correct ones as we move towards 1.0, then that is what we have to do. `ButtonComponent` is what we should have. `ButtonGroup.Buttons` should be made up of those components.